FirebaseInAppMessaging/Swift/Tests/Unit/FirebaseInAppMessaging_APIBuildTests.swift

```diff
@@ -0,0 +1,309 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+
+import FirebaseInAppMessaging
+import FirebaseInAppMessagingSwift
+import SwiftUI
+
+final class FirebaseInAppMessagingSwift_APIBuildTests: XCTestCase {
+  func usage() throws {
+    // MARK: - FirebaseInAppMessaging
+
+    let inAppMessaging = FirebaseInAppMessaging.InAppMessaging.inAppMessaging()
+
+    let _: Bool = inAppMessaging.messageDisplaySuppressed
+    inAppMessaging.messageDisplaySuppressed = true
+
+    // TODO(ncooke3): This should probably be removed in favor of the one in
+    // FirebaseCore?
+    let _: Bool = inAppMessaging.automaticDataCollectionEnabled
+    inAppMessaging.automaticDataCollectionEnabled = true
+
+    let _: FirebaseInAppMessaging.InAppMessagingDisplay = inAppMessaging.messageDisplayComponent
+    // FIXME(ncooke3): Below.
+    let displayConformer: FirebaseInAppMessaging.InAppMessagingDisplay! = nil
+    inAppMessaging.messageDisplayComponent = displayConformer
+
+    inAppMessaging.triggerEvent("eventName")
+
+    let delegate: FirebaseInAppMessaging.InAppMessagingDisplayDelegate? = inAppMessaging.delegate
+    inAppMessaging.delegate = nil
+
+    // TODO(ncooke3): Does it make sense for these params to be nullable?
+    let nullableText: String? = nil
+    let nullableURL: URL? = nil
+    let action = FirebaseInAppMessaging.InAppMessagingAction(
+      actionText: nullableText,
+      actionURL: nullableURL
+    )
+    let _: String? = action.actionText
+    let _: URL? = action.actionURL
+
+    let nonnullText = ""
+    let nonnullColor: UIColor = .black
+    let button = FirebaseInAppMessaging.InAppMessagingActionButton(
+      buttonText: nonnullText,
+      buttonTextColor: nonnullColor,
+      backgroundColor: nonnullColor
+    )
+    let _: String = button.buttonText
+    let _: UIColor = button.buttonTextColor
+    let _: UIColor = button.buttonBackgroundColor
+
+    // TODO(ncooke3): This type should not have a `FIR` prefix.
+    _ = FirebaseInAppMessaging.FIRInAppMessagingDisplayMessageType.RawValue()
+    let messsageType: FirebaseInAppMessaging.FIRInAppMessagingDisplayMessageType! = nil
+    switch messsageType! {
+    case .modal: break
+    case .banner: break
+    case .imageOnly: break
+    case .card: break
+    @unknown default: break
```

Comment:
What's the value of `@unknown` here? Wouldn't a build break be better than a warning if we add something new?

Reply:
It will throw a warning if a case is added to the API. Without the `@unknown`, adding a new case to the API would  not produce a warning. I can take it away, but we will get the following warning:
<img width="1122" alt="Screenshot 2023-09-11 at 4 38 15 PM" src="https://github.com/firebase/firebase-ios-sdk/assets/36927374/23733d6b-ffe2-480c-8c91-de50e98974c4">

Something we can explore in the future would be making our public enums frozen (can mark up with `NS_CLOSED_ENUM` in ObjC) to prevent clients from having to do deal with this warning.

Reply:
I was reading up on this here: https://www.swiftbysundell.com/articles/using-an-unknown-default-case-within-a-switch-statement/

Reply:
Hmm. I'm confused. The first sentence from that link: "When switching on an enum, we are required to either explicitly handle all of its cases".  How can there be additional unknown values? The compiler can see the FIRInAppMessagingDismissType's definition.

Reply:
I see. I think it should have said _When switching on a~~n~~ [frozen] enum, we are required to either explicitly handle all of its cases_

Reply:
I think this article is a bit clearer actually: https://www.avanderlee.com/swift/unknown-default-enums-in-swift/

Reply:
From the article, this only affect enums defined in a C-Family language: https://www.avanderlee.com/swift/unknown-default-enums-in-swift/#do-i-ever-need-the-unknown-case-with-my-own-enums

Reply:
Yes.